### ISSUE TITLE
Bug: [consistent-type-imports] does not fix to specified style

### ISSUE THREAD

**Issue by simPod** [typescript-eslint/typescript-eslint#9652]
### Before You File a Bug Report Please Confirm You Have Done The Following...

- [X] I have tried restarting my IDE and the issue persists.
- [X] I have updated to the latest version of the packages.
- [X] I have [searched for related issues](https://github.com/typescript-eslint/typescript-eslint/issues?q=is%3Aissue+label%3A%22package%3A+eslint-plugin%22) and found none that matched my issue.
- [X] I have [read the FAQ](https://typescript-eslint.io/linting/troubleshooting) and my problem is not listed.

### Playground Link

https://typescript-eslint.io/play/#ts=5.5.2&fileType=.ts&code=JYWwDg9gTgLgBAbzjAnmApnAYhCcC%2BcAZlBCHAOQ4QUDcAUKJLMmpgEICGUxp5FXKHXqoMcACpwAvNlwMAxhAB2AZ3gAPAFxxB0uAEZaQA&eslintrc=N4KABGBEBOCuA2BTAzpAXGUEKQAIBcBPABxQGNoBLY-AWhXkoDt8B6MgeyeUuX0Ra0ipWpQC2xDtHyoMAbXDYciaNCmQANIqVYl2SADNKADwDKRJOijJExAIbQ7-ISUSiJUmZG3YAvj4BdRX9fIA&tsconfig=&tokens=false

### Repro Code

```TypeScript
import { type Foo } from 'Foo';
```

### ESLint Config

```javascript
{
  "rules": {
    "@typescript-eslint/consistent-type-imports": [
      "error",
      {
        "fixStyle": "separate-type-imports"
      }
    ]
  }
}¨``
```

### tsconfig

```jsonc
{
  "compilerOptions": {
    // ...
  }
}
```

### Expected Result

I expect the rule to fix the type prefix:

`import type { Foo } from 'Foo';`

```diff
- import { type Foo } from 'Foo';
+ import type { Foo } from 'Foo';
```

### Actual Result

no fix

**Comment by bradzacher** [comment]
You misunderstand.
The rule doesn't enforce your imports are in a specified style. It just fixes from a value to a type import with the specified style.

Please see the rule docs for other rules that can do the checks you want.

**Comment by simPod** [comment]
For everyone else, this seems to work @typescript-eslint/no-import-type-side-effects

**Comment by bradzacher** [comment]
This would likely be the better rule to enforce a consistent type specifier style in your codebase, as its name suggests
https://github.com/import-js/eslint-plugin-import/blob/main/docs/rules/consistent-type-specifier-style.md
